Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -2685,31 +2685,41 @@ Ref<DownloadProxy> WebsiteDataStore::createDownloadProxy(Ref<API::DownloadClient
     return protectedNetworkProcess()->createDownloadProxy(*this, WTFMove(client), request, frameInfo, originatingPage);
 }
 
+void WebsiteDataStore::prepareRequestForDownload(WebCore::ResourceRequest& request, WebPageProxy* originatingPage)
+{
+    // Request's firstPartyForCookies will be used as Original URL of the download request.
+    // We set the value to top level document's URL.
+    if (originatingPage) {
+        URL originatingPageURL = URL { originatingPage->pageLoadState().url() };
+        request.setFirstPartyForCookies(originatingPageURL);
+        request.setIsSameSite(WebCore::areRegistrableDomainsEqual(originatingPageURL, request.url()));
+        if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
+        request.setHTTPUserAgent(originatingPage->userAgentForURL(request.url()));
```

Comment:
```suggestion
        if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
            request.setHTTPUserAgent(originatingPage->userAgentForURL(request.url()));
```

it's odd the style checker didn't pick this